### PR TITLE
Greener CI

### PR DESCRIPTION
**Pull request by TriplEight**
- remove the compile error
- `clippy` suggestions

This is kind of my first Rust experience on such project, and I couldn't get rid of all the warnings, but I tried really hard xD
So please review.

### CODE REVIEW

File: src/compiler/args.rs [hunk 1]
```diff
@@ -357,7 +357,7 @@ impl IntoArg for PathBuf {
         self.into()
     }
     fn into_arg_string(self, transformer: PathTransformerFn<'_>) -> ArgToStringResult {
-        transformer(&self).ok_or_else(|| ArgToStringError::FailedPathTransform(self))
+        transformer(&self).ok_or(ArgToStringError::FailedPathTransform(self))
```
Comment (drahnr) [inline src/compiler/args.rs:360]: I would leave this as a closure

File: src/compiler/compiler.rs [hunk 2]
```diff
@@ -1401,10 +1398,7 @@ LLVM version: 6.0",
             }))
             .unwrap();
         // Ensure that the object file was created.
-        assert_eq!(
-            true,
-            fs::metadata(&obj).and_then(|m| Ok(m.len() > 0)).unwrap()
-        );
+        assert_eq!(true, fs::metadata(&obj).map(|m| m.len() > 0).unwrap());
```
Comment (drahnr) [inline src/compiler/compiler.rs:1401]: We should rewrite this as an `assert_eq(Ok(1),...)`

File: src/compiler/compiler.rs [hunk 3]
```diff
@@ -1438,10 +1432,7 @@ LLVM version: 6.0",
...
```
Comment (drahnr) [inline src/compiler/compiler.rs:1435]: CC

File: src/compiler/args.rs (hunk 1 above)
Comment (drahnr) [inline src/compiler/args.rs:360]:
```suggestion
        transformer(&self).ok_or_else(|| ArgToStringError::FailedPathTransform(self))
```

Comment (drahnr) [inline src/compiler/args.rs]:
```suggestion
         transformer(&self).ok_or_else(|| ArgToStringError::FailedPathTransform(self))
```

**Review by drahnr — APPROVED** [review]

File: src/compiler/args.rs (hunk 1 above)
Comment (TriplEight) [reply src/compiler/args.rs]: done

File: src/compiler/compiler.rs (hunk 2 above)
Comment (TriplEight) [reply src/compiler/compiler.rs:1403]: I didn't get it, it's Result(Integer) x bool

**Review by drahnr — DISMISSED** [review]

File: tests/harness/mod.rs [hunk 4]
```diff
@@ -128,8 +128,7 @@ pub fn sccache_client_cfg(tmpdir: &Path) -> sccache::config::FileConfig {
     fs::create_dir(tmpdir.join(cache_relpath)).unwrap();
     fs::create_dir(tmpdir.join(dist_cache_relpath)).unwrap();
 
-    let mut disk_cache: sccache::config::DiskCacheConfig = Default::default();
-    disk_cache.dir = tmpdir.join(cache_relpath);
+    let disk_cache: sccache::config::DiskCacheConfig = Default::default();
```
Comment (drahnr) [inline tests/harness/mod.rs:131]:
```suggestion
    let disk_cache: sccache::config::DiskCacheConfig = { 
         dir: { tmpdir.join(cache_relpath) },
        .. Default::default()
     };
```

**Review by drahnr — APPROVED** [review]

**Comment by drahnr** [comment]
Must be **squashed**

**Pull request merged**